app/controllers/doorkeeper/tokens_controller.rb

```diff
@@ -37,5 +37,9 @@ def revoke_token(token)
     def strategy
       @strategy ||= server.token_request params[:grant_type]
     end
+
+    def authorize_response
+      @authorize_response ||= strategy.authorize
```

Comment:
Actually this is the main reason I sent this pull request, because `strategy.authorize` is not idempotent, if in a subclass of `TokensController` I need the response, I cannot call `strategy.authorize` again, therefore the need for the memoization.